web/tests/message_edit.test.cjs

```diff
@@ -4,15 +4,26 @@ const assert = require("node:assert/strict");
 
 const {mock_esm, zrequire} = require("./lib/namespace.cjs");
 const {run_test} = require("./lib/test.cjs");
+const $ = require("./lib/zjquery.cjs");
 
 const message_edit = zrequire("message_edit");
 const people = zrequire("people");
+const stream_popover = zrequire("stream_popover");
 const {set_current_user, set_realm} = zrequire("state_data");
 
 const is_content_editable = message_edit.is_content_editable;
 
 const settings_data = mock_esm("../src/settings_data");
 const stream_data = mock_esm("../src/stream_data");
+const message_util = mock_esm("../src/message_util");
+const message_lists = mock_esm("../src/message_lists");
+const narrow_state = mock_esm("../src/narrow_state");
+
+message_lists.current = {
+    data: {
+        fetch_status: {},
+    },
+};
```

Comment:
In general this test uses a lot more mocking than would be ideal. Better would be to just popular objects in `stream_data` using the APIs in `web/tests/lib/example_stream.ts`, and similarly for other objects.

But it's still usable.

Reply:
Actually, I take that back -- I'm not convinced your tests here are actually testing any of the logic that we care about here for correctness -- it seems like you're mocking the code that would be most likely to be wrong in each case. 

I recommend (1) adding your new tests in a new test file, `message_move_counts.test.cjs`, and (2) writing that test without using `mock_esm` -- just create objects using the relevant APIs, so that we're actually exercising the real code.